pages/api/flashcards/[id].js

```diff
@@ -38,13 +41,18 @@ export default async function handler(request, response) {
 
   if (request.method === "PATCH") {
     try {
-      const updatedFlashcard = request.body;
-      if (!updatedFlashcard) {
-        response.status(404).json({ status: "Not Found" });
+      if (session) {
+        const updatedFlashcard = request.body;
+        if (!updatedFlashcard) {
+          response.status(404).json({ status: "Not Found" });
+          return;
+        }
+        await Flashcard.findByIdAndUpdate(id, updatedFlashcard);
+        response.status(200).json({ message: "Flashcard updated." });
+      } else {
+        response.status(401).json({ status: "Not authorized" });
```

Comment:
this pattern that you are using is called nesting if statements. it can get difficult to read after a couple of layers. You can use early return statements instead that function like security checks before the relevant code:
```suggestion
      if (!session) {
        response.status(401).json({ status: "Not authorized" });
        return 
      }
      
      const updatedFlashcard = request.body;
      if (!updatedFlashcard) {
        response.status(404).json({ status: "Not Found" });
        return;
      }
      
      await Flashcard.findByIdAndUpdate(id, updatedFlashcard);
      response.status(200).json({ message: "Flashcard updated." });       
```
by this you have one layer of nesting less and the code is more readable.

Reply:
All right, I changed the spelling everywhere in the backend so that I now set an early return in the `!session` query.